compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -965,7 +965,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             }
 
             // `T` -> `Trait`
-            (_, &ty::Dynamic(ref data, r, ty::Dyn)) => {
+            (_, &ty::Dynamic(ref data, r, _kind)) => {
```

Comment:
This is suspicious. Can you explain what unsizing we need this for?

Reply:
Maybe the issue is higher up, like we are mistakenly considering a cast to `dyn*` as an unsizing.

I interpreted this as "here's the check you have to do to cast to `dyn Trait`" and the obligations there look like the same ones we want for `dyn*` (except that we'll want to add a `PointerSized` requirement also once that PR merges).

Reply:
Removed this in favor of #103386.